### DISCUSSION TITLE
How to programmatically call the Column-Picker Dialog Ok button

### DISCUSSION THREAD

**Discussion by ibrahimGit123** [serenity-is/Serenity#6226] [Help]
On my Grid, I dynamically add and remove columns. However, this usually causes issues with my grid. Refreshing the grid does not fix the issue. I find that the only way to fix the issue is to manually open the Column-Picker Dialog and click "Ok".

Is there a way to simulate clicking the column-picker "OK" button programmatically in the grid.ts file? Or a way to programmatically perform the same action the button performs on the Slick grid?

**Comment by wezmag** [comment]
I found this blog article written by @minhhungit 
https://minhhungit.github.io/2020/10/07/010-adding-slickgrid-column-in-serenity-dynamically/

**Reply by ibrahimGit123** [reply]
Thanks - however I tried this and it did not work.

**Reply by wezmag** [reply]
This is what happened when Ok button clicked.
https://github.com/serenity-is/Serenity/blob/523c0195fec17089e1dd7e812650c2edaf47a7ac/src/Serenity.Scripts/CoreLib/UI/DataGrid/ColumnPickerDialog.ts#L120-L141

Line 140, it also called `this.done()`, which is define here.
https://github.com/serenity-is/Serenity/blob/523c0195fec17089e1dd7e812650c2edaf47a7ac/src/Serenity.Scripts/CoreLib/UI/DataGrid/ColumnPickerDialog.ts#L60-L66

I think that's all you need to fix the issue.

**Reply by ibrahimGit123** [reply]
Thanks - this is what I was looking for
